src/browser/fetch/method.rs

```diff
@@ -0,0 +1,37 @@
+/// HTTP Method types.
+///
+/// [https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods)
+#[derive(Debug, Clone, Copy)]
+pub enum Method {
+    Get,
+    Head,
+    Post,
+    Put,
+    Delete,
+    Connect,
+    Options,
+    Trace,
+    Patch,
+}
+
+impl Method {
+    pub fn as_str(self) -> &'static str {
+        match self {
+            Method::Get => "GET",
+            Method::Head => "HEAD",
+            Method::Post => "POST",
+            Method::Put => "PUT",
+            Method::Delete => "DELETE",
+            Method::Connect => "CONNECT",
+            Method::Options => "OPTIONS",
+            Method::Trace => "TRACE",
+            Method::Patch => "PATCH",
```

Comment:
`Method` vs `Self`

Reply:
Hm?

Reply:
e.g. `Self::Head => "HEAD"` instead of `Method::Head => "HEAD"`. I'm surprised that Clippy is ok wih it.

Reply:
The same for `impl Default`.